x-pack/plugin/sql/sql-proto/src/main/java/org/elasticsearch/xpack/sql/proto/SqlVersions.java

```diff
@@ -0,0 +1,219 @@
+    public static final SqlVersion V_8_11_2 = fromId(8_11_02_99);
+    public static final SqlVersion V_8_11_3 = fromId(8_11_03_99);
+    public static final SqlVersion V_8_11_4 = fromId(8_11_04_99);
+    public static final SqlVersion V_8_12_0 = fromId(8_12_00_99);
+    public static final SqlVersion V_8_12_1 = fromId(8_12_01_99);
+    public static final SqlVersion V_8_12_2 = fromId(8_12_02_99);
+    public static final SqlVersion V_8_13_0 = fromId(8_13_00_99);
+    public static final SqlVersion V_8_13_1 = fromId(8_13_01_99);
+    public static final SqlVersion V_8_13_2 = fromId(8_13_02_99);
+    public static final SqlVersion V_8_13_3 = fromId(8_13_03_99);
+    public static final SqlVersion V_8_13_4 = fromId(8_13_04_99);
+    public static final SqlVersion V_8_14_0 = fromId(8_14_00_99);
+    public static final SqlVersion V_8_14_1 = fromId(8_14_01_99);
+    public static final SqlVersion V_8_14_2 = fromId(8_14_02_99);
+    public static final SqlVersion V_8_14_3 = fromId(8_14_03_99);
+    public static final SqlVersion V_8_15_0 = fromId(8_15_00_99);
+    public static final SqlVersion V_8_15_1 = fromId(8_15_01_99);
+    public static final SqlVersion V_8_16_0 = fromId(8_16_00_99);
+
+    static final List<SqlVersion> DECLARED_VERSIONS = getDeclaredVersions();
+
+    /**
+     * What's the version of the server that the clients should be compatible with?
+     */
+    public static final SqlVersion SERVER_COMPAT_VERSION = getLatestVersion();
+
+    public static SqlVersion getFirstVersion() {
+        return DECLARED_VERSIONS.get(0);
+    }
+
+    public static SqlVersion getLatestVersion() {
+        return DECLARED_VERSIONS.get(DECLARED_VERSIONS.size() - 1);
+    }
+
+    public static SqlVersion getPreviousVersion(SqlVersion version) {
+        int index = Collections.binarySearch(DECLARED_VERSIONS, version);
+        if (index < 1) {
+            throw new IllegalArgumentException("couldn't find any released versions before [" + version + "]");
+        }
+        return DECLARED_VERSIONS.get(index - 1);
+    }
+
+    public static SqlVersion getNextVersion(SqlVersion version) {
+        int index = Collections.binarySearch(DECLARED_VERSIONS, version);
+        if (index >= DECLARED_VERSIONS.size() - 1) {
+            throw new IllegalArgumentException("couldn't find any released versions before [" + version + "]");
+        }
+        return DECLARED_VERSIONS.get(index + 1);
+    }
+
+    public static List<SqlVersion> getAllVersions() {
+        return DECLARED_VERSIONS;
+    }
+
+    // lifted from org.elasticsearch.Version#getDeclaredVersions
+    private static List<SqlVersion> getDeclaredVersions() {
+        final Field[] fields = SqlVersions.class.getFields();
+        final List<SqlVersion> versions = new ArrayList<>(fields.length);
+        for (final Field field : fields) {
+            final int mod = field.getModifiers();
+            if (false == Modifier.isStatic(mod) && Modifier.isFinal(mod) && Modifier.isPublic(mod)) {
```

Comment:
Shouldn't this be 
```suggestion
            if (false == (Modifier.isStatic(mod) && Modifier.isFinal(mod) && Modifier.isPublic(mod))) {
```

Reply:
Good catch. Seems like the structure of SqlVersion (and Version, where this was gotten from) made the check safe, for now, but fixed it, thanks.